provider/gce/google/errors.go

```diff
@@ -102,6 +103,15 @@ func HasDenialStatusCode(err error) bool {
 			}
 		}
 	}
+	if cause, ok := errors.Cause(err).(*googleapi.Error); ok {
```

Comment:
You can merge these two loops together - they just need cause to be an `error`.

Something like:
```go
var cause error
switch e := errors.Cause(err).(type) {
case *urlError:
    cause = e
case *googleapi.Error:
    cause := e
default: 
    return false
}
for code, descs := range AuthorisationFailureStatusCodes {
     ...
}
return false
```

Reply:
?